src/basic/time-util.c

```diff
@@ -1486,7 +1486,7 @@ int time_change_fd(void) {
 
         /* We only care for the cancellation event, hence we set the timeout to the latest possible value. */
         static const struct itimerspec its = {
-                .it_value.tv_sec = TIME_T_MAX,
+                .it_value.tv_sec = INT_MAX,
```

Comment:
The lines below where there is a `#if SIZEOF_TIME_T == 8 && ULONG_MAX < UINT64_MAX ... #endif` block claims to handle this case. Is it nor working?